pkg/util/heap/heap.go

```diff
@@ -169,8 +170,9 @@ func (h *Heap) Len() int {
 	return h.data.Len()
 }
 
-// List returns a list of all the items.
+// List returns a sorted list of all the items.
 func (h *Heap) List() []interface{} {
+	sort.Sort(&h.data)
```

Comment:
I don't think we want to sort here. First, IIUC, it can destroy the state of the heap. Second, the sorting should take into account both the heap.items and the inadmissibleWorkloads. Finally, it is preferable to make the sorting and post processing outside of the read lock, to make it as quick as possible.